Makefile

```diff
@@ -33,23 +39,46 @@ pre-pr: tidy generate lint test
 clean:
 	-@rm test/testdata/x.wasm
 
-sdk_srcs := $(wildcard ./basm/**/*.go)
-wasm_src_dir := $(shell realpath ./test/testdata)
-wasm_src := $(wasm_src_dir)/main.go
-wasm_out := $(wasm_src_dir)/x.wasm
+sdk_src := $(wildcard ./basm/**/*.go)
+wasm_src := $(testdata_dir)/main.go
+wasm_out := $(testdata_dir)/x.wasm
 
 .PHONY: wasm
 wasm: $(wasm_out)
 
-$(wasm_out): $(wasm_src) $(sdk_srcs)
+$(wasm_out): $(wasm_src) $(sdk_src)
 	@echo "Building WASM module..."
 	@tinygo build -o $@ -target=wasi $<
 
-test_scripts := $(shell realpath $(wildcard ./test/scripts/*.txtar))
+# Set default values for BASM_ variables. These can be overridden by the user
+# e.g. `BASM_USER_SECRET=mysecret make test`
+BASM_PLATFORM ?= plain
+BASM_CODE_MEASURE ?= plain
+BASM_AUTH_TOKEN ?= auth token
+BASM_HOST ?= local-server
+BASM_USER_SECRET ?= this is a bearer token
+# Export all BASM_ variables for use in recipes
+export $(filter BASM_%,$(.VARIABLES))
+
+template_src := $(wildcard $(template_dir)/*.tmpl)
+
+template_out := $(patsubst $(template_dir)/%.tmpl,$(rendered_template_dir)/%,$(template_src))
+# Mark the rendered files as intermediate. Make will delete them after the build.
+.INTERMEDIATE: $(template_out)
+
+# Rule to render template files, `envsubst <template_file> > <output_file>`
+$(rendered_template_dir)/%: $(template_dir)/%.tmpl
+	@envsubst < $< > $@
```

Comment:
envsubst has one job: accept an input, expand $VAR or ${VAR} values using environment variable values, and send the rendered input to stdout.

Reply:
I wasn't sure about introducing a new tool, but the benefit here is the templates will populate all values automatically from the environment, rather than having to pass a set of named arguments. 

What do you think? I like how simple the tool is, but I don't think it would be that much of a stretch to use mustache here (I already had to name all of the env vars explicitly above, so the value of not having to pass in args explicitly has somewhat been lost) if we want to keep using the familiar tool mustache.

Reply:
I moved to using jq and mustache